frontend/src/__mocks__/mockDscStatus.ts

```diff
@@ -49,7 +51,23 @@ export const mockDscStatus = ({
   phase = 'Ready',
   release = { name: 'Open Data Hub', version: '2.28.0' },
 }: MockDscStatus): DataScienceClusterKindStatus => ({
-  components,
+  components: (() => {
+    // Map StackComponent -> DataScienceStackComponent ComponentName keys
+
+    const result = { ...components };
+    // Synthesize components managementState from installedComponents for compatibility
+    Object.entries(installedComponents).forEach(([stackKey, isInstalled]) => {
+      const statusKey = stackToStatusKey[stackKey as StackComponent];
+      if (statusKey) {
+        const prev = result[statusKey] || {};
+        result[statusKey] = {
+          managementState: isInstalled ? 'Managed' : 'Removed',
```

Comment:
nit: I see several places in which we use `'Managed'` and `'Removed'`, should we consider creating an enum for that?